src/common/lib/lnurl.ts

```diff
@@ -61,27 +68,46 @@ const lnurl = {
 
     return null;
   },
+
   async getDetails(lnurlString: string): Promise<LNURLError | LNURLDetails> {
     const url = normalizeLnurl(lnurlString);
-    let lnurlDetails = {} as LNURLDetails;
-    lnurlDetails.tag = url.searchParams.get("tag") as LNURLDetails["tag"];
-    if (lnurlDetails.tag === "login") {
-      lnurlDetails.k1 = url.searchParams.get("k1") || "";
-      lnurlDetails.action = url.searchParams.get("action") || undefined;
+    const searchParamsTag = url.searchParams.get("tag");
+    const searchParamsK1 = url.searchParams.get("k1");
+    const searchParamsAction = url.searchParams.get("action");
+
+    if (searchParamsTag && searchParamsTag === "login" && searchParamsK1) {
+      const lnurlAuthDetails: LNURLAuthServiceResponse = {
+        ...(searchParamsAction && { action: searchParamsAction }),
+        domain: url.hostname,
+        k1: searchParamsK1,
+        tag: searchParamsTag,
+        url: url.toString(),
+      };
+
+      return lnurlAuthDetails;
     } else {
       try {
-        const res = await axios.get(url.toString());
-        lnurlDetails = res.data;
+        const { data }: { data: LNURLDetails | LNURLError } = await axios.get(
+          url.toString()
+        );
+        const lnurlDetails = data;
+
+        if (!isLNURLDetailsError(lnurlDetails)) {
```

Comment:
hmmm, how do we deal with errors when it is an LNURLDetailsError?

Reply:
My understanding is that this is just a typeguard to avoid TS complaining. If the axios request fails it should fall into the catch (like the code before).  

@lisabaut wdyt?

Reply:
this is not for when the HTTP request fails. this checking the lnurl response: 
https://github.com/getAlby/lightning-browser-extension/blob/6c3cd1012e63f6ad3c0a206f56e90f04c5d204f0/src/common/utils/typeHelpers.ts#L6

the HTTP request would already fail in line 90 and end up in the catch block.

Reply:
Got it! Adjusted it